File: demo/src/main/java/com/ahk/arg/ForSale/models/entities/operations/Venta.java
package com.ahk.arg.ForSale.models.entities.operations;

import com.ahk.arg.ForSale.models.entities.persona.Empleado;

public class Venta extends Operacion{

    private static Float porcentajeComision = 1.5F;

    @Override
    public Float calcularComisionPara(Empleado empleado) {
        return super.inmueble.precioFinal() * porcentajeComision;
    }
}

File: entities/venta.js
import { Operacion } from "./operacion";

export class Venta extends Operacion{
    porcentaje;

    constructor(inmueble){
        super("venta", inmueble);
        this.comision = this.calcularComision();
    }

    calcularComision(){
        const PORCENTAJE = 1.5;
        PORCENTAJE/100 * this.inmueble.valor;
    }

}